### PR TITLE
created climb method

### PR DESCRIPTION
**Pull request by OfriRosenbaum**

### CODE REVIEW

File: src/main/java/frc/robot/Commands/Climb.java [hunk 1]
```diff
@@ -1,4 +1,25 @@
...
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import frc.robot.Subsystems.ClimberPlacer;
+import frc.robot.Subsystems.ClimberWinch;
+
+public class Climb extends SequentialCommandGroup {
+
+    public Climb() {
+        ClimberWinch climberWinch = ClimberWinch.getInstance();
+        ClimberPlacer climberPlacer = ClimberPlacer.getInstance();
+        addRequirements(climberWinch, climberPlacer);
```
Comment (tal-sitton) [inline src/main/java/frc/robot/Commands/Climb.java:13]: why?

File: src/main/java/frc/robot/Commands/Climb.java [hunk 2]
```diff
@@ -1,4 +1,25 @@
...
+import com.spikes2212.command.genericsubsystem.commands.MoveGenericSubsystem;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import frc.robot.Subsystems.ClimberPlacer;
+import frc.robot.Subsystems.ClimberWinch;
+
+public class Climb extends SequentialCommandGroup {
+
+    public Climb() {
+        ClimberWinch climberWinch = ClimberWinch.getInstance();
+        ClimberPlacer climberPlacer = ClimberPlacer.getInstance();
```
Comment (tal-sitton) [inline src/main/java/frc/robot/Commands/Climb.java:12]: remove

File: src/main/java/frc/robot/Subsystems/ClimberPlacer.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package frc.robot.Subsystems;
```
Comment (tal-sitton) [inline src/main/java/frc/robot/Subsystems/ClimberPlacer.java:1]: remove

File: src/main/java/frc/robot/Subsystems/ClimberWinch.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+import com.spikes2212.command.genericsubsystem.MotoredGenericSubsystem;
+import edu.wpi.first.wpilibj.motorcontrol.MotorController;
+
+import java.util.function.Supplier;
+
+public class ClimberWinch extends MotoredGenericSubsystem {
+
+    private Supplier<Double> UP_SPEED = rootNamespace.addConstantDouble("up speed", 0.6);
+    private Supplier<Double> DOWN_SPEED = rootNamespace.addConstantDouble("down speed", -0.4);
```
Comment (tal-sitton) [inline src/main/java/frc/robot/Subsystems/ClimberWinch.java:11]: 0.4 is wayyyy too much

**Review by tal-sitton — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/Subsystems/ClimberWinch.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
+package frc.robot.Subsystems;
+
+import com.spikes2212.command.genericsubsystem.MotoredGenericSubsystem;
+import edu.wpi.first.wpilibj.motorcontrol.MotorController;
+
+import java.util.function.Supplier;
+
+public class ClimberWinch extends MotoredGenericSubsystem {
+
+    private Supplier<Double> UP_SPEED = rootNamespace.addConstantDouble("up speed", 0.6);
```
Comment (Yoel-Perman-Brilliant) [inline src/main/java/frc/robot/Subsystems/ClimberWinch.java:10]: UP_SPEED, DOWN_SPEED, HOOKED_DOWN_SPEED can all be made public and final, which makes their respective getters redundant

There's a good chance I have no idea what I'm talking about though

**Review by Yoel-Perman-Brilliant — COMMENTED** [review]
I'm so sorry

File: src/main/java/frc/robot/Commands/Climb.java [hunk 6]
```diff
@@ -1,4 +1,23 @@
...
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import frc.robot.Subsystems.ClimberWinch;
+
+public class Climb extends SequentialCommandGroup {
+
+    public Climb() {
+        ClimberWinch climberWinch = ClimberWinch.getInstance();
+        addRequirements(climberWinch);
+        addCommands(
+                new MoveGenericSubsystem(climberWinch, climberWinch.getUpSpeed()),
```
Comment (tal-sitton) [inline src/main/java/frc/robot/Commands/Climb.java:13]: this won't work, like, physically.
you need to move closer to the bar after you open the telescopic arm for this system to catch it.

**Review by tal-sitton — CHANGES_REQUESTED** [review]

**Review by tal-sitton — APPROVED** [review]

**Review by tal-sitton — APPROVED** [review]

**Pull request merged**